File: docs/examples/cpp/src/lifecycle_policy_create.cc
#include <cassert>
#include <reduct/client.h>

using reduct::Error;
using reduct::IClient;

int main() {
  auto client = IClient::Build("http://127.0.0.1:8383", {.api_token = "my-token"});

  auto [bucket, bucket_err] = client->GetOrCreateBucket("my-bucket");
  assert(bucket_err == Error::kOk);

  auto err = client->CreateLifecycle("my-lifecycle", IClient::LifecycleSettings{
      .type = IClient::LifecycleType::kDelete,
      .bucket = "my-bucket",
      .entries = {"cpp-example"},
      .max_age = "30d",
      .interval = "1h",
      .when = R"({"&anomaly":{"$eq":1}})",
  });
  assert(err == Error::kOk);
}

File: versioned_docs/version-1.21.x/examples/go/src/lifecycle_policy_create.go
package main

import (
	"context"

	reduct "github.com/reductstore/reduct-go"
	model "github.com/reductstore/reduct-go/model"
)

func main() {
	client := reduct.NewClient("http://localhost:8383", reduct.ClientOptions{APIToken: "my-token"})
	_, err := client.CreateOrGetBucket(context.Background(), "my-bucket", nil)
	if err != nil {
		panic(err)
	}

	settings := model.LifecycleSettings{
		LifecycleType: model.LifecycleTypeDelete,
		Bucket:        "my-bucket",
		Entries:       []string{"go-example"},
		OlderThan:     "30d",
		Interval:      "1h",
		When: map[string]any{
			"&anomaly": map[string]any{"$eq": 1},
		},
	}

	err = client.CreateLifecycle(context.Background(), "my-lifecycle", settings)
	if err != nil {
		panic(err)
	}
}
